src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[10];
+    }
+
     @Override
     public void add(T value) {
-
+        array[size] = value;
+        addSize();
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkCorrectIndex(index);
+        addSize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkCorrectIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkCorrectIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkCorrectIndex(index);
+        return removeElement(index);
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length - 1; i++) {
+            if (array[i] != null && array[i].equals(element)) {
```

Comment:
add point when array[i] = null, like this: 'if (array[i] != null && array[i].equals(element) || array[i] == element) {'